pkg/kube/util/boshdns/corefile.go

```diff
@@ -42,14 +42,18 @@ type HandlerSource struct {
 	Type      string   `json:"type"`
 }
 
-// HTTP returns true if the handler's source is of type http
-func (h HandlerSource) HTTP() bool {
-	return h.Type == "http"
-}
-
-// DNS returns true if the handler's source is of type dns
-func (h HandlerSource) DNS() bool {
-	return h.Type == "dns"
+// Protocol returns the coredns server protocol.
+func (h HandlerSource) Protocol() string {
+	switch h.Type {
+	case "http", "tls":
```

Comment:
I thought 'http' for 'tls' appeared in the original story and was used by BOSH somehow, but can't find it anymore. Maybe we should drop this, as it is confusing and just support the Coredns settings (dns, tls, https, grpc).